core/src/main/java/org/ldaptive/AbstractConnectionValidator.java

```diff
@@ -55,6 +64,38 @@ public void setValidateTimeout(final Duration timeout)
   }
 
 
+  @Override
+  public Boolean apply(final Connection conn)
+  {
+    if (conn == null) {
+      return false;
+    }
+    return applyAsync(conn).get();
+  }
+
+
+  @Override
+  public Supplier<Boolean> applyAsync(final Connection conn)
+  {
+    final BlockingQueue<Boolean> queue = new ArrayBlockingQueue<>(1);
```

Comment:
Use of a blocking queue to synchronize on a single boolean value struck me as practical but inefficient, so I took some time to see if I could come up with a more efficient alternative. Sent you a PR for my solution: https://github.com/vt-middleware/ldaptive/pull/190.

Reply:
PR merged.